File: python/realtime_raw_point_by_vins.py
from client import create_gateway_client, retry_stream, TIMEOUT_SEC
import compassiot.platform.v1.streaming_pb2 as streaming


def main():
    client = create_gateway_client()

    request = streaming.RealtimeRawPointByVinsRequest(
        vins=[],
        stream_env= streaming.StreamEnvironment.DEV
    )
    
    for response in retry_stream(lambda: client.RealtimeRawPointByVins(request, timeout=TIMEOUT_SEC)):
        print(response)


if __name__ == "__main__":
    main()

File: typescript/src/realtime_raw_point_by_vins.ts
import * as streaming from "@buf/compassiot_api.bufbuild_es/compassiot/platform/v1/streaming_pb"

import { TIMEOUT_MS, createNodeClient, retryStream } from "./client"

const client = createNodeClient()

const request = new streaming.RealtimeRawPointByVinsRequest({
  vins: [],
  streamEnv: streaming.StreamEnvironment.DEV
})

for await (const response of retryStream(() => client.realtimeRawPointByVins(request, { timeoutMs: TIMEOUT_MS }))) {
  console.log(response.toJsonString({ prettySpaces: 2 }))
}
